services/motion/builtin/builtin.go

```diff
@@ -348,9 +348,6 @@ func (ms *builtIn) PlanHistory(
 //     input value: a motionplan.Trajectory
 //     output value: a bool
 func (ms *builtIn) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
-	ms.mu.Lock()
-	defer ms.mu.Unlock()
```

Comment:
so no more mutex at all?
how is this safe?

Reply:
We no longer need a full lock but you're right, this should have a read lock